js/sky/src/grids/columnpicker.js

```diff
@@ -77,6 +83,44 @@
         $scope.locals.selectedColumns = initialSelectedColumns.slice(0);
         $scope.columnPickerHelpKey = columnPickerHelpKey;
         $scope.listMode = listMode;
+        if (colPickerSearchProperties) {
+            $scope.colPickerSearchProperties = colPickerSearchProperties;
+            $scope.searchCallback = function (args) {
+                var i,
+                    j, 
+                    n,
+                    m, 
+                    p,
+                    val,
+                    item, 
+                    items = $scope.initialColumns,
+                    filteredItems = [],
+                    searchTextUpper = angular.isString(args.searchText) ? args.searchText.toUpperCase() : '';
+                for (i = 0, n = items.length; i < n; i++) {
+                    item = items[i];
+                        
+                    if (!args.category || item.category === args.category) {
+                        if (!searchTextUpper) {
+                            filteredItems.push(item);
+                        } else {
+                            for (j = 0, m = $scope.colPickerSearchProperties.length; j < m; j++) {
+                                p = $scope.colPickerSearchProperties[j];
+                                if (item.hasOwnProperty(p)) {
+                                    val = item[p];
+
+                                    if (angular.isString(val) && val.toUpperCase().indexOf(searchTextUpper) >= 0) {
+                                        filteredItems.push(item);
+                                        break;
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+                
+                $scope.columns = translateColumnsToChecklistObjects(filteredItems); 
+            };
+        }
```

Comment:
I think your concerns are valid, but I'm not sure how much benefit we would actually get out of that. The columnpicker searching is modeled off of the checklist behavior, but there isn't very much actual exact code reuse between the two of them now (the itemMatchesCategory is the only immediately obvious one to me). As long as the bb-checklist-filter-callback behavior doesn't change, they wouldn't need to even update the grid code as long as the tests still pass and functionality stays the same.